plutus-use-cases/src/Plutus/Contracts/Uniswap/OffChain.hs

```diff
@@ -346,7 +346,8 @@ add us AddParams{..} = do
 
         tx       = Constraints.mustPayToTheScript dat val          <>
                    Constraints.mustMintValue lVal                  <>
-                   Constraints.mustSpendScriptOutput oref redeemer
+                   Constraints.mustSpendScriptOutput oref redeemer <>
+                   Constraints.mustIncludeDatum (Datum $ PlutusTx.toBuiltinData dat)
```

Comment:
I don't remember the exact chain of responsibility here, but IIRC if we don't include the datum explicitly, it's passed as a reference, and it's not included in the `txInfo` built by cardano-node (that relies on `makeTransactionBody'`)

Reply:
In this case, because you're using `mustPayTheScript` the datum is added automatically in the tx body. It's only in the `mustPayToOtherScript` that the datum is not added automatically.

Reply:
I removed it and it passes, I think it was necessary in the middle in the PR and fixed later, so it can be removed now.

Reply:
You haven't commited the change I think :)